metal/v1/model_address.go

```diff
@@ -286,6 +287,29 @@ func (o Address) ToMap() (map[string]interface{}, error) {
 }
 
 func (o *Address) UnmarshalJSON(bytes []byte) (err error) {
+	// This validates that all required properties are included in the JSON object
+	// by unmarshalling the object into a generic map with string keys and checking
+	// that every required field exists as a key in the generic map.
+	requiredProperties := []string{
+		"address",
+		"country",
+		"zip_code",
+	}
+
+	allProperties := make(map[string]interface{})
+
+	err = json.Unmarshal(bytes, &allProperties)
+
+	if err != nil {
+		return err
+	}
+
+	for _, requiredProperty := range requiredProperties {
+		if _, exists := allProperties[requiredProperty]; !exists {
+			return fmt.Errorf("no value given for required property %v", requiredProperty)
+		}
+	}
+
```

Comment:
This is an example of the required property validation that is added in v7.1.0.  Models with required properties are updated to check that the required property names appear in the JSON object that is being unmarshalled.